File: tests/src/com.typesafe.akka/akka-actor_2.13/2.8.8/src/test/scala/com_typesafe_akka/akka_actor_2_13/LineNumbersTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_actor_2_13

import akka.util.LineNumbers
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class LineNumbersTest {

  @Test
  def readsSourceInformationFromAnOrdinaryObjectClassResource(): Unit = {
    val target: PlainLineNumberTarget = new PlainLineNumberTarget("ordinary-object")

    val result: LineNumbers.Result = LineNumbers(target)

    assertThat(result).isNotEqualTo(LineNumbers.NoSourceInfo)
    assertThat(result).isNotInstanceOf(classOf[LineNumbers.UnknownSourceFormat])
    assertThat(result.toString).contains("LineNumbersTest.scala")
  }

  @Test
  def readsSourceInformationFromASerializableLambdaImplementationClassResource(): Unit = {
    val target: SerializableJavaLambdaFactory.SerializableCallable =
      SerializableJavaLambdaFactory.create("serializable-lambda")

    assertThat(target.call()).isEqualTo("serializable-lambda")
    val result: LineNumbers.Result = LineNumbers(target.asInstanceOf[AnyRef])

    assertThat(result).isNotEqualTo(LineNumbers.NoSourceInfo)
    assertThat(result).isNotInstanceOf(classOf[LineNumbers.UnknownSourceFormat])
    assertThat(result.toString).contains("SerializableJavaLambdaFactory.java")
  }

  private final class PlainLineNumberTarget(val name: String)
}

File: tests/src/org.sonatype.sisu/sisu-guice/2.1.7/src/test/java/org_sonatype_sisu/sisu_guice/LineNumbersTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_sonatype_sisu.sisu_guice;

import static org.assertj.core.api.Assertions.assertThatThrownBy;

import com.google.inject.ConfigurationException;
import com.google.inject.Guice;
import com.google.inject.Inject;
import org.junit.jupiter.api.Test;

public class LineNumbersTest {
    @Test
    void formatsMissingConstructorDependencyWithSourceLineNumbers() {
        assertThatThrownBy(() -> Guice.createInjector().getInstance(ConstructorTarget.class))
                .isInstanceOf(ConfigurationException.class)
                .hasMessageContaining("No implementation for " + MissingService.class.getName())
                .hasMessageContaining("for parameter 0 at")
                .hasMessageContaining("LineNumbersTest.java");
    }

    public static class ConstructorTarget {
        @Inject
        public ConstructorTarget(MissingService service) {
        }
    }

    public interface MissingService {
    }
}
